contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -416,7 +369,17 @@ abstract contract AbstractTokenConverter is AccessControlledV8, IAbstractTokenCo
     ) public returns (uint256 amountConvertedMantissa, uint256 amountOutMantissa) {
         priceOracle.updateAssetPrice(tokenAddressIn);
         priceOracle.updateAssetPrice(tokenAddressOut);
-        (amountConvertedMantissa, amountOutMantissa) = getAmountOut(amountInMantissa, tokenAddressIn, tokenAddressOut);
+        (amountConvertedMantissa, amountOutMantissa, ) = _getAmountOut(
+            amountInMantissa,
+            tokenAddressIn,
+            tokenAddressOut
+        );
+        uint256 maxTokenOutReserve = balanceOf(tokenAddressOut);
+
+        /// If contract has less liquity for tokenAddressOut than amountOutMantissa
+        if (maxTokenOutReserve < amountOutMantissa) {
+            revert InsufficientPoolLiquidity();
+        }
```

Comment:
Why don't we do the same adjustment that we are doing in `getAmountOut`? It seems inconsistent